src/test/java/com/fasterxml/jackson/databind/deser/BeanDeserializerModifier4216Test.java

```diff
@@ -0,0 +1,62 @@
+package com.fasterxml.jackson.databind.deser;
+
+import static com.fasterxml.jackson.databind.BaseMapTest.jsonMapperBuilder;
+import static junit.framework.TestCase.assertEquals;
```

Comment:
Can use you use org.junit.Asserts?

Reply:
or even use the Jupiter asserts, but put the params in the right order - the ones expected by the Jupiter (actual, expected)

Reply:
Ah, thank you for pointing out! Totally missed that.

Reply:
I was confused. So looked into `org.Junit.Assert.assertEquals()` seems to be declared in `exepected, actual` order, actually 🤔, or could you lemme know if there is another Assert?

<img width="669" alt="image" src="https://github.com/FasterXML/jackson-databind/assets/61615301/3787a2be-a66d-45d1-98f8-55a35ee02811">